src/convertToObject.js

```diff
@@ -7,6 +7,21 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const rules = {};
+
+  sourceString
+    .trim()
+    .split(';')
+    .forEach((rule) => {
+      const [key, value] = rule
+        .trim()
+        .split(':')
+        .map((part) => part.trim());
+
+      rules[key] = value;
+    });
```

Comment:
[CODE PRACTICE] - Instead of using the `forEach` loop, you should use more functional approaches like `reduce` to directly create the resulting object from the array of rules. This adheres to the checklist item that suggests avoiding loops and using iteration methods instead.